Active transform feedback must affect useProgram

lexaknyazev opened the pull request:
#### f1fd6ea39943fbd147877d7e5ea9bc359d9a4911
<pre>
Active transform feedback must affect useProgram
<a href="https://bugs.webkit.org/show_bug.cgi?id=242632">https://bugs.webkit.org/show_bug.cgi?id=242632</a>

Reviewed by Kenneth Russell.

Extended WebGLRenderingContextBase::useProgram flow
to check for an active and not paused transform feedback.

Added a helper function to WebGL2RenderingContext.

Fixes forthcoming strengthened conformance test in
<a href="https://github.com/KhronosGroup/WebGL/pull/3462">https://github.com/KhronosGroup/WebGL/pull/3462</a>

* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::isTransformFeedbackActiveAndUnpaused):
* Source/WebCore/html/canvas/WebGL2RenderingContext.h:
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
(WebCore::WebGLRenderingContextBase::useProgram):

Canonical link: <a href="https://commits.webkit.org/252432@main">https://commits.webkit.org/252432@main</a>
</pre>

kenrussell reviewed (approved):
I'm ambivalent about requiring that all of the validation checks happen in the same order - in Chromium I've preferred to override useProgram and add the WebGL 2.0-specific validation there. But regardless this looks fine.

In this PR's description, please mention:

"""
Fixes forthcoming strengthened conformance test in
https://github.com/KhronosGroup/WebGL/pull/3462 .
"""

Thanks.

lexaknyazev committed:
Active transform feedback must affect useProgram
https://bugs.webkit.org/show_bug.cgi?id=242632

Reviewed by Kenneth Russell.

Extended WebGLRenderingContextBase::useProgram flow
to check for an active and not paused transform feedback.

Added a helper function to WebGL2RenderingContext.

Fixes forthcoming strengthened conformance test in
https://github.com/KhronosGroup/WebGL/pull/3462

* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::isTransformFeedbackActiveAndUnpaused):
* Source/WebCore/html/canvas/WebGL2RenderingContext.h:
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
(WebCore::WebGLRenderingContextBase::useProgram):

Canonical link: https://commits.webkit.org/252432@main
